packages/typespec-vscode/src/openapi3-preview.ts

```diff
@@ -0,0 +1,251 @@
+import { readdir, readFile, rm } from "fs/promises";
+import { basename, dirname, join } from "path";
+import * as semver from "semver";
+import * as vscode from "vscode";
+import logger from "./log/logger.js";
+import { TspLanguageClient } from "./tsp-language-client.js";
+import { createTempDir, ExecOutput } from "./utils.js";
+
+export async function getMainTspFile(): Promise<string | undefined> {
+  const files = await vscode.workspace.findFiles("**/main.tsp").then((uris) => {
+    return uris.filter((uri) => uri.scheme === "file" && !uri.fsPath.includes("node_modules"));
+  });
+
+  switch (files.length) {
+    case 0:
+      const errMsg = "No 'main.tsp' file found in the workspace.";
+      logger.error(errMsg);
+      vscode.window.showErrorMessage(errMsg);
+      return undefined;
+    case 1:
+      return files[0].fsPath;
+    default:
+      return await vscode.window.showQuickPick(
+        files.map((file) => file.fsPath),
+        {
+          title: "Select the 'main.tsp' file",
```

Comment:
I think we are using "title" for scenario while "placeholder" for step detail. Please double check with Josephine about the title and placeholder.

Reply:
updated to follow the convention